src/lib/components/LeafletMap/components/Controls/components/MousePosition.js

```diff
@@ -296,11 +296,17 @@ const MousePosition = props => {
 
     const mapZValue = redColorvalue => {
         const { minvalue, maxvalue } = stateRef.current || {};
-        return Math.floor(
+        return roundOff(
             minvalue + ((maxvalue - minvalue) / (255 - 0)) * (redColorvalue - 0)
         );
     };
 
+    const roundOff = n => {
+        return parseFloat(
+            n.toExponential(Math.max(1, 2 + Math.log10(Math.abs(n))))
+        );
+    };
+
```

Comment:
Add docstring/comments explaining what this function does (since it is not easy to read from the equation)?